package/libretro-snes9x2002/0002-Add-snesadvance.patch

```diff
@@ -0,0 +1,770 @@
++EF15F4C3|Yoshi's Cookie (U)|0|0|0|0|0|0|2824F=421B
++D138F224|Yoshi's Island (V1.0) (U)|0|0|0|0|0|0|F4=423B
++CF98DDAA|Yoshi's Island (U) (M3) (V1.1)|F4=423B
++59490CE8|Yoshi's Safari|0|0|0|0|0|0
++7EA1AFE8|Young Merlin|0|0|0|0|0|0
++64A91E64|Ys 3 - Wanderers from Ys|0|0|0|0|0|0|D8=42
++1ADD47F0|Ys 4 - Mask of the Sun (J) [T-E|4|0|0|0|0|0|14060C=EAEA,140631=EAEA,9CA=42DC,A3B=42DC
++7B4CEBAF|Ys 4 - Mask of the Sun (J) [T-E|4|0|0|0|0|14060C|14060C=EAEA,140631=EAEA,9CA=42DC,A3B=42DC
++CFE632B9|Ys V - Expert (J)|4|44800|0|0|0|0|886=EAEA,79D=EAEA,3B81=EAEA,86C=EAEA,5E06=42FB,A9C=42FB,5DCC=42DB
++EC96D517|YUUYUU Hakusho (J)|0|0|0|0|0|0|E15=42FC
++EEF45A93|YUUYUU Hakusho 2 - Kakutou no S|0|0|0|0|0|0|10F=42DC
++5617A42E|YUUYUU Hakusho Final (J)|1|0|0|0|0|0|A71D=EAEA,A683=EAEA,AB=42DC
++535CCCAC|YUUYUU Hakusho Tokubehuten (J)|0|0|0|0|0|0|7957=EAEA,781A=EAEA
++5409D4F4|Zen Nihon GT Sensyuken (J)|1320000|0|0|0|0|0|8003B=EAEA,8012B=EAEA,80137=EAEA,15E=42FB
++5397D5BC|Zero the Kamikaze Squirrel (U)|0|0|0|0|0|0|41B44=EAEA,41B69=EAEA,1F3=42DB,12EA=4214,2884A=42DB,496=421B,41F=421B
++7CFC0C7C|Zombies Ate My Neighbors (U)|10000000|4040|0|0|0|0|4CCF=EAEA,9DC=42
+diff --git a/libretro/libretro.c b/libretro/libretro.c
+index 3d863eb..f819378 100644
+--- a/libretro/libretro.c
++++ b/libretro/libretro.c
+@@ -645,6 +645,16 @@ static void check_variables(bool first_run)
+          two_c = 3;
+       }
+    }
++   
++   var.key = "snes9x2002_snesadvance_hacks";
++   var.value = NULL;
++
++   Settings.SpeedHacks = true;
++
++   if (environ_cb(RETRO_ENVIRONMENT_GET_VARIABLE, &var) && var.value)
++      if (strcmp(var.value, "disabled") == 0)
++         Settings.SpeedHacks = false;
++
+ 
+    /* Reinitialise frameskipping, if required */
+    if (!first_run &&
+diff --git a/libretro/libretro_core_options.h b/libretro/libretro_core_options.h
+index ad03eae..9fdf58a 100644
+--- a/libretro/libretro_core_options.h
++++ b/libretro/libretro_core_options.h
+@@ -194,6 +194,20 @@ struct retro_core_option_v2_definition option_defs_us[] = {
+          { NULL, NULL },
+       },
+       "disabled"
++   },
++      {
++      "snes9x2002_snesadvance_hacks",
++      "Speed hacks from snesadvance.dat file",
++      NULL,
++      "Enable speed hacks from snesadvance.dat file on ROM load (Restart required)",
++      NULL,
++      NULL,
++      {
++         { "disabled", NULL },
++         { "enabled",  NULL },
++         { NULL, NULL },
++      },
++      "enabled"
+    },
+    { NULL, NULL, NULL, NULL, NULL, NULL, {{0}}, NULL },
+ };
+diff --git a/src/memmap.c b/src/memmap.c
+index a66a5e6..994dfd6 100644
+--- a/src/memmap.c
++++ b/src/memmap.c
+@@ -301,6 +301,70 @@ void FreeSDD1Data ()
+     }
+ }
+ 
++int os9x_findhacks(int game_crc32){
++	int i=0,j;
++	int _crc32;	
++	char c;
++	char str[256];
++	unsigned int size_snesadvance;
++	unsigned char *snesadvance;
++	FILE *f;
++	
++	sprintf(str,"%s/snesadvance.dat","/mnt/.retroarch/system/snes9x2002");

```

Comment:
if you want to specify system dir use maybe ifdef __LIBRETRO__ and just call RETRO_ENVIRONMENT_GET_SYSTEM_DIRECTORY to make it look cleaner